addons/l10n_th/views/account_move.xml

```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="report_invoice_document" inherit_id="account.report_invoice_document">
+        <xpath expr="//div[@name='address_not_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice">
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
+
+        <xpath expr="//div[@name='address_same_as_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice" >
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
+        <xpath expr="//div[@name='no_shipping']//span[@t-field='o.partner_id.vat']" position="after">
+            <t t-if="o.company_id.account_fiscal_country_id.code == 'TH' and not commercial_invoice" >
+                <span t-esc="o.partner_id.l10n_th_branch_name"/>
+            </t>
+        </xpath>
```

Comment:
This is painful :weary: 
Could be a lot easier if `report_invoice_document` was refactored a tiny bit... Probably to do in `master`

Reply:
What do you suggest is the best approach for this case?